### Thesis Project Code

# Obtaining strain-level resolution of key species of chicken microbiome and the geographical distribution of these.

This repositiory was created in relation to my thesis project working with metagenomic data. 

The main part of the project was run on a computer cluster and all the commandlines run there can be found in the codelog.txt file. 

### Overview
The output from different programs run on the computer cluster are found in some of the folders (FastQC_reports,  Roary_output, assembly_stat_plots, drep_output_figures, inStrain_output, Kaiju_analysis, nonpareil), but a few are in the data56_related_files folder (CheckM_results_file, Preprocessing_output, gtdbtk.bac120.summary.tsv). 

The output have been used in different analysis of which the most are found within the Jupyter_Notebooks. Folders nonpareil, kaiju_analysis and assembly_stat_plots contain the R code for the respective analysis and visulaisatioins. 

The resultFigs contains the output figures from the notebooks while the result from the R code are within their respective folders. 


### Acknowledgements  
A great thanks to my supervisors Gisle Alberg Vestergaard and Lea Benedict Skov Hansen. 