{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "40c72e92",
   "metadata": {},
   "source": [
    "## Examining the GTDB output\n",
    "\n",
    "This notebook does the following:\n",
    "\n",
    "- Checking if clusters contian multiple species, genus, or families\n",
    "- generate dataframes containing all the resulting annotations and on for only Lactobacillaceae\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "89057e26",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sn \n",
    "import numpy as np\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a1de1dd",
   "metadata": {},
   "source": [
    "#### Open GTDBtk results (bac120andAr122.summary.tsv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "88c98a27",
   "metadata": {},
   "outputs": [],
   "source": [
    "f = open(\"../data56_related_files/gtdbtk.bac_ar_summary_relevant_columns.tsv\", \"r\")\n",
    "files = f.read()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0796ca87",
   "metadata": {},
   "source": [
    "#### Create list and dataframe "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "acd7c03e",
   "metadata": {},
   "outputs": [],
   "source": [
    "fileList = files.split(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "82df6bbf",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(fileList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a22876b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>user_genome\\tclassification\\tclosest_placement...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S10C1032\\td__Bacteria;p__Bacteroidota;c__Bacte...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S10C1039\\td__Bacteria;p__Firmicutes;c__Bacilli...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S10C1113\\td__Bacteria;p__Firmicutes;c__Bacilli...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S10C119\\td__Bacteria;p__Firmicutes;c__Bacilli;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2380</th>\n",
       "      <td>user_genome\\tclassification\\tclosest_placement...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2381</th>\n",
       "      <td>S23C1277\\td__Archaea;p__Methanobacteriota;c__M...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2382</th>\n",
       "      <td>S34C1277\\td__Archaea;p__Methanobacteriota;c__M...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2383</th>\n",
       "      <td>S53C3468\\td__Archaea;p__Methanobacteriota;c__M...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2384</th>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2385 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      0\n",
       "0     user_genome\\tclassification\\tclosest_placement...\n",
       "1     S10C1032\\td__Bacteria;p__Bacteroidota;c__Bacte...\n",
       "2     S10C1039\\td__Bacteria;p__Firmicutes;c__Bacilli...\n",
       "3     S10C1113\\td__Bacteria;p__Firmicutes;c__Bacilli...\n",
       "4     S10C119\\td__Bacteria;p__Firmicutes;c__Bacilli;...\n",
       "...                                                 ...\n",
       "2380  user_genome\\tclassification\\tclosest_placement...\n",
       "2381  S23C1277\\td__Archaea;p__Methanobacteriota;c__M...\n",
       "2382  S34C1277\\td__Archaea;p__Methanobacteriota;c__M...\n",
       "2383  S53C3468\\td__Archaea;p__Methanobacteriota;c__M...\n",
       "2384                                                   \n",
       "\n",
       "[2385 rows x 1 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f549180a",
   "metadata": {},
   "source": [
    "#### Cleaning df and splitting string to columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "85e59e45",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.columns = ['temp']\n",
    "# user_genome\tclassification\tfastani_reference\tfastani_reference_radius\tfastani_taxonomy\t\n",
    "# fastani_ani\tfastani_af\tclosest_placement_reference\tclosest_placement_radius\tclosest_placement_taxonomy\t\n",
    "# closest_placement_ani\tclosest_placement_af\tpplacer_taxonomy\tclassification_method\tnote\t\n",
    "# other_related_references(genome_id,species_name,radius,ANI,AF)\tmsa_percent\ttranslation_table\tred_value\n",
    "# warnings\n",
    "\n",
    "df[['user_genome','classification', 'closest_placement_taxonomy',\n",
    "    'msa_percent']] = df.temp.str.split(\"\\t\",expand=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3c315e53",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>temp</th>\n",
       "      <th>user_genome</th>\n",
       "      <th>classification</th>\n",
       "      <th>closest_placement_taxonomy</th>\n",
       "      <th>msa_percent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>user_genome\\tclassification\\tclosest_placement...</td>\n",
       "      <td>user_genome</td>\n",
       "      <td>classification</td>\n",
       "      <td>closest_placement_taxonomy</td>\n",
       "      <td>msa_percent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S10C1032\\td__Bacteria;p__Bacteroidota;c__Bacte...</td>\n",
       "      <td>S10C1032</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>33.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S10C1039\\td__Bacteria;p__Firmicutes;c__Bacilli...</td>\n",
       "      <td>S10C1039</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>94.58</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                temp  user_genome  \\\n",
       "0  user_genome\\tclassification\\tclosest_placement...  user_genome   \n",
       "1  S10C1032\\td__Bacteria;p__Bacteroidota;c__Bacte...     S10C1032   \n",
       "2  S10C1039\\td__Bacteria;p__Firmicutes;c__Bacilli...     S10C1039   \n",
       "\n",
       "                                      classification  \\\n",
       "0                                     classification   \n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "2  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "\n",
       "                          closest_placement_taxonomy  msa_percent  \n",
       "0                         closest_placement_taxonomy  msa_percent  \n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...        33.75  \n",
       "2  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...        94.58  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bf9fa282",
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop first row because we get double headers \n",
    "df = df.drop(0)\n",
    "\n",
    "# drop the last row since it is empty\n",
    "last_row = len(df)\n",
    "df = df.drop(df.index[last_row-1]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "63d71bf1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop first coulmn that is the original combined string\n",
    "df = df.iloc[: , 1:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8c161b5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Because archea and bacteria were concatinated the header line is present further down where the archea file starts. \n",
    "df.drop(index=df[df['user_genome'] == 'user_genome'].index, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "95b7f417",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>user_genome</th>\n",
       "      <th>classification</th>\n",
       "      <th>closest_placement_taxonomy</th>\n",
       "      <th>msa_percent</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>sample</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S10C1032</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>33.75</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Bacteroidaceae</td>\n",
       "      <td>g__Phocaeicola</td>\n",
       "      <td>s__Phocaeicola dorei</td>\n",
       "      <td>S10</td>\n",
       "      <td>1032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S10C1039</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>94.58</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>s__Limosilactobacillus ingluviei</td>\n",
       "      <td>S10</td>\n",
       "      <td>1039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S10C1113</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>96.07</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>s__Limosilactobacillus oris</td>\n",
       "      <td>S10</td>\n",
       "      <td>1113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S10C119</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>67.12</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Weissella</td>\n",
       "      <td>s__Weissella thailandensis</td>\n",
       "      <td>S10</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>S10C1295</td>\n",
       "      <td>d__Bacteria;p__Campylobacterota;c__Campylobact...</td>\n",
       "      <td>d__Bacteria;p__Campylobacterota;c__Campylobact...</td>\n",
       "      <td>44.93</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Campylobacterota</td>\n",
       "      <td>c__Campylobacteria</td>\n",
       "      <td>o__Campylobacterales</td>\n",
       "      <td>f__Helicobacteraceae</td>\n",
       "      <td>g__Helicobacter_D</td>\n",
       "      <td>s__Helicobacter_D pullorum</td>\n",
       "      <td>S10</td>\n",
       "      <td>1295</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  user_genome                                     classification  \\\n",
       "1    S10C1032  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "2    S10C1039  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "3    S10C1113  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "4     S10C119  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "5    S10C1295  d__Bacteria;p__Campylobacterota;c__Campylobact...   \n",
       "\n",
       "                          closest_placement_taxonomy msa_percent       Domain  \\\n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...       33.75  d__Bacteria   \n",
       "2  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       94.58  d__Bacteria   \n",
       "3  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       96.07  d__Bacteria   \n",
       "4  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       67.12  d__Bacteria   \n",
       "5  d__Bacteria;p__Campylobacterota;c__Campylobact...       44.93  d__Bacteria   \n",
       "\n",
       "                Phylum               Class                 Order  \\\n",
       "1      p__Bacteroidota      c__Bacteroidia      o__Bacteroidales   \n",
       "2        p__Firmicutes          c__Bacilli    o__Lactobacillales   \n",
       "3        p__Firmicutes          c__Bacilli    o__Lactobacillales   \n",
       "4        p__Firmicutes          c__Bacilli    o__Lactobacillales   \n",
       "5  p__Campylobacterota  c__Campylobacteria  o__Campylobacterales   \n",
       "\n",
       "                 Family                   Genus  \\\n",
       "1     f__Bacteroidaceae          g__Phocaeicola   \n",
       "2   f__Lactobacillaceae  g__Limosilactobacillus   \n",
       "3   f__Lactobacillaceae  g__Limosilactobacillus   \n",
       "4   f__Lactobacillaceae            g__Weissella   \n",
       "5  f__Helicobacteraceae       g__Helicobacter_D   \n",
       "\n",
       "                            Species sample cluster  \n",
       "1              s__Phocaeicola dorei    S10    1032  \n",
       "2  s__Limosilactobacillus ingluviei    S10    1039  \n",
       "3       s__Limosilactobacillus oris    S10    1113  \n",
       "4        s__Weissella thailandensis    S10     119  \n",
       "5        s__Helicobacter_D pullorum    S10    1295  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8b147627",
   "metadata": {},
   "outputs": [],
   "source": [
    "#splitting classification to columns based on tax-rank\n",
    "df[['Domain','Phylum', 'Class', 'Order', 'Family', \n",
    "    'Genus', 'Species']] = df.classification.str.split(\";\",expand=True)\n",
    "#splitting user_genome ID to sample and cluster\n",
    "df[['sample','cluster']] = df.user_genome.str.split(\"C\",expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "7b3d261f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>user_genome</th>\n",
       "      <th>classification</th>\n",
       "      <th>closest_placement_taxonomy</th>\n",
       "      <th>msa_percent</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>sample</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S10C1032</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>33.75</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Bacteroidaceae</td>\n",
       "      <td>g__Phocaeicola</td>\n",
       "      <td>s__Phocaeicola dorei</td>\n",
       "      <td>S10</td>\n",
       "      <td>1032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S10C1039</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>94.58</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>s__Limosilactobacillus ingluviei</td>\n",
       "      <td>S10</td>\n",
       "      <td>1039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S10C1113</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>96.07</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>s__Limosilactobacillus oris</td>\n",
       "      <td>S10</td>\n",
       "      <td>1113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S10C119</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>67.12</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Weissella</td>\n",
       "      <td>s__Weissella thailandensis</td>\n",
       "      <td>S10</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>S10C1295</td>\n",
       "      <td>d__Bacteria;p__Campylobacterota;c__Campylobact...</td>\n",
       "      <td>d__Bacteria;p__Campylobacterota;c__Campylobact...</td>\n",
       "      <td>44.93</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Campylobacterota</td>\n",
       "      <td>c__Campylobacteria</td>\n",
       "      <td>o__Campylobacterales</td>\n",
       "      <td>f__Helicobacteraceae</td>\n",
       "      <td>g__Helicobacter_D</td>\n",
       "      <td>s__Helicobacter_D pullorum</td>\n",
       "      <td>S10</td>\n",
       "      <td>1295</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  user_genome                                     classification  \\\n",
       "1    S10C1032  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "2    S10C1039  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "3    S10C1113  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "4     S10C119  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "5    S10C1295  d__Bacteria;p__Campylobacterota;c__Campylobact...   \n",
       "\n",
       "                          closest_placement_taxonomy msa_percent       Domain  \\\n",
       "1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...       33.75  d__Bacteria   \n",
       "2  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       94.58  d__Bacteria   \n",
       "3  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       96.07  d__Bacteria   \n",
       "4  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       67.12  d__Bacteria   \n",
       "5  d__Bacteria;p__Campylobacterota;c__Campylobact...       44.93  d__Bacteria   \n",
       "\n",
       "                Phylum               Class                 Order  \\\n",
       "1      p__Bacteroidota      c__Bacteroidia      o__Bacteroidales   \n",
       "2        p__Firmicutes          c__Bacilli    o__Lactobacillales   \n",
       "3        p__Firmicutes          c__Bacilli    o__Lactobacillales   \n",
       "4        p__Firmicutes          c__Bacilli    o__Lactobacillales   \n",
       "5  p__Campylobacterota  c__Campylobacteria  o__Campylobacterales   \n",
       "\n",
       "                 Family                   Genus  \\\n",
       "1     f__Bacteroidaceae          g__Phocaeicola   \n",
       "2   f__Lactobacillaceae  g__Limosilactobacillus   \n",
       "3   f__Lactobacillaceae  g__Limosilactobacillus   \n",
       "4   f__Lactobacillaceae            g__Weissella   \n",
       "5  f__Helicobacteraceae       g__Helicobacter_D   \n",
       "\n",
       "                            Species sample cluster  \n",
       "1              s__Phocaeicola dorei    S10    1032  \n",
       "2  s__Limosilactobacillus ingluviei    S10    1039  \n",
       "3       s__Limosilactobacillus oris    S10    1113  \n",
       "4        s__Weissella thailandensis    S10     119  \n",
       "5        s__Helicobacter_D pullorum    S10    1295  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "a5929a50",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.cluster = df.cluster.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "765ae961",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('../data56_related_files/GTDB_All.csv', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a1c0eab",
   "metadata": {},
   "source": [
    "## Sanity Check\n",
    "Check if the bins in the same cluster has recieved the same taxonomical assignation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "c4d0ff22",
   "metadata": {},
   "outputs": [],
   "source": [
    "def CheckingIdenticalTaxInCluster_all(df):\n",
    "    '''\n",
    "    identify clusters containing different species\n",
    "    1. group by cluster\n",
    "    2. identify/log unique species, where species are more than one - exclude s__\n",
    "    3. if species is undefined - s__ , identify unique genus for s__, where genus are more than one\n",
    "    4. if genus is undefined, identify unique family, where family are more than one'''\n",
    "\n",
    "    df.sort_values(by=['cluster'], inplace=True)\n",
    "    f = open(\"data56_related_files/log_cluster_irregularities.csv\", \"a\")\n",
    "    \n",
    "    grouped = df.groupby('cluster')\n",
    "\n",
    "    for index, c in grouped:\n",
    "        cluster = c['cluster'].unique()\n",
    "        \n",
    "        uniqueGenus = []\n",
    "        uniqueFamily = []\n",
    "        \n",
    "        # value in cluster is unique\n",
    "        uniqueSpecies = c['Species'].unique()\n",
    "    \n",
    "        if 's__' in uniqueSpecies:\n",
    "            # check genus\n",
    "            uniqueGenus = c['Genus'].unique()\n",
    "            if 'g__' in uniqueGenus:\n",
    "                # check family\n",
    "                uniqueFamily = c['Family'].unique()\n",
    "\n",
    "\n",
    "        \n",
    "        if (len(uniqueSpecies) > 1 or len(uniqueGenus) > 1 or len(uniqueFamily) > 1):\n",
    "            \n",
    "            if (len(np.delete(uniqueSpecies, np.where(uniqueSpecies == \"s__\"))) < 2 and \n",
    "                len(np.delete(uniqueGenus, np.where(uniqueGenus == \"g__\"))) < 2 and len(uniqueFamily) < 2):\n",
    "                continue\n",
    "                \n",
    "            # there is more than one - now log it\n",
    "            f.write(f\"Species: {uniqueSpecies}, Genus: {uniqueGenus}, Family: {uniqueFamily}, Cluster: {cluster} \\n\")\n",
    "        \n",
    "    f.close()\n",
    "    \n",
    "CheckingIdenticalTaxInCluster_all(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "1f9779bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !rm data56_related_files/log_cluster_irregularities.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "f749be4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>user_genome</th>\n",
       "      <th>classification</th>\n",
       "      <th>closest_placement_taxonomy</th>\n",
       "      <th>msa_percent</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>sample</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>727</th>\n",
       "      <td>S27C1</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>N/A</td>\n",
       "      <td>11.12</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Coprobacteraceae</td>\n",
       "      <td>g__</td>\n",
       "      <td>s__</td>\n",
       "      <td>S27</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1868</th>\n",
       "      <td>S50C1</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>15.39</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Barnesiellaceae</td>\n",
       "      <td>g__Barnesiella</td>\n",
       "      <td>s__</td>\n",
       "      <td>S50</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>659</th>\n",
       "      <td>S26C1</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>N/A</td>\n",
       "      <td>19.0</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Coprobacteraceae</td>\n",
       "      <td>g__</td>\n",
       "      <td>s__</td>\n",
       "      <td>S26</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1543</th>\n",
       "      <td>S45C1</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>N/A</td>\n",
       "      <td>10.56</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Coprobacteraceae</td>\n",
       "      <td>g__</td>\n",
       "      <td>s__</td>\n",
       "      <td>S45</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     user_genome                                     classification  \\\n",
       "727        S27C1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "1868       S50C1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "659        S26C1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "1543       S45C1  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "\n",
       "                             closest_placement_taxonomy msa_percent  \\\n",
       "727                                                 N/A       11.12   \n",
       "1868  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...       15.39   \n",
       "659                                                 N/A        19.0   \n",
       "1543                                                N/A       10.56   \n",
       "\n",
       "           Domain           Phylum           Class             Order  \\\n",
       "727   d__Bacteria  p__Bacteroidota  c__Bacteroidia  o__Bacteroidales   \n",
       "1868  d__Bacteria  p__Bacteroidota  c__Bacteroidia  o__Bacteroidales   \n",
       "659   d__Bacteria  p__Bacteroidota  c__Bacteroidia  o__Bacteroidales   \n",
       "1543  d__Bacteria  p__Bacteroidota  c__Bacteroidia  o__Bacteroidales   \n",
       "\n",
       "                   Family           Genus Species sample  cluster  \n",
       "727   f__Coprobacteraceae             g__     s__    S27        1  \n",
       "1868   f__Barnesiellaceae  g__Barnesiella     s__    S50        1  \n",
       "659   f__Coprobacteraceae             g__     s__    S26        1  \n",
       "1543  f__Coprobacteraceae             g__     s__    S45        1  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['cluster'] == 1]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64df9f30",
   "metadata": {},
   "source": [
    "Number of undefined at each rank:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "24364d9e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1093"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df[\"Species\"] == 's__'].count()['user_genome']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "90f4731b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "108"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df[\"Genus\"] == 'g__'].count()['user_genome']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "296c681a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df[\"Family\"] == 'f__'].count()['user_genome']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "aad136d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "user_genome                   2382\n",
       "classification                 467\n",
       "closest_placement_taxonomy     397\n",
       "msa_percent                   1786\n",
       "Domain                           2\n",
       "Phylum                          17\n",
       "Class                           21\n",
       "Order                           52\n",
       "Family                         100\n",
       "Genus                          246\n",
       "Species                        269\n",
       "sample                          56\n",
       "cluster                       1067\n",
       "dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eca22aa3",
   "metadata": {},
   "source": [
    "### Overview\n",
    "Out of 2382 bins/rows \n",
    "\n",
    "* 1093 bins with species undefined \n",
    "* 108 bins with genus undefined \n",
    "* 0 bins with families undefined\n",
    "\n",
    "Out of 1067 clusters \n",
    "* 2 times did species differ in two clusters - s__Corynebacterium pollutisoli/s__Corynebacterium xerosis, and s__Escherichia flexneri/s__Escherichia coli_D in cluster 456 and 672\n",
    "* 4 times did genus differ within a cluster \n",
    "* 6 times did families differ within a cluster - in all cases the confusion was between f__Coprobacteraceae and f__Barnesiellaceae. \n",
    "\n",
    "see more in the log_clusters_irregularities file. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d22a0ce",
   "metadata": {},
   "source": [
    "### Creating GTDB dataframe for Lactobacillaceae bins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "4d570b56",
   "metadata": {},
   "outputs": [],
   "source": [
    "Lactobacillaceae_df =df[df[\"Family\"] == 'f__Lactobacillaceae']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "7c7f1bc1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "Lactobacillaceae_df = pd.DataFrame(Lactobacillaceae_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "8edc259b",
   "metadata": {},
   "outputs": [],
   "source": [
    "Lactobacillaceae_df.to_csv('../data56_related_files/GTDB_Lactobacillaceae.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3f02f7a6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>user_genome</th>\n",
       "      <th>classification</th>\n",
       "      <th>closest_placement_taxonomy</th>\n",
       "      <th>msa_percent</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>sample</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1545</th>\n",
       "      <td>S45C105</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>40.0</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>s__</td>\n",
       "      <td>S45</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>643</th>\n",
       "      <td>S25C105</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>73.26</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>s__</td>\n",
       "      <td>S25</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1782</th>\n",
       "      <td>S49C105</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>53.94</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>s__</td>\n",
       "      <td>S49</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1322</th>\n",
       "      <td>S39C105</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>65.52</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Limosilactobacillus</td>\n",
       "      <td>s__</td>\n",
       "      <td>S39</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2250</th>\n",
       "      <td>S8C119</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>77.29</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Bacilli</td>\n",
       "      <td>o__Lactobacillales</td>\n",
       "      <td>f__Lactobacillaceae</td>\n",
       "      <td>g__Weissella</td>\n",
       "      <td>s__Weissella thailandensis</td>\n",
       "      <td>S8</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     user_genome                                     classification  \\\n",
       "1545     S45C105  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "643      S25C105  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "1782     S49C105  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "1322     S39C105  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "2250      S8C119  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "\n",
       "                             closest_placement_taxonomy msa_percent  \\\n",
       "1545  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...        40.0   \n",
       "643   d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       73.26   \n",
       "1782  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       53.94   \n",
       "1322  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       65.52   \n",
       "2250  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...       77.29   \n",
       "\n",
       "           Domain         Phylum       Class               Order  \\\n",
       "1545  d__Bacteria  p__Firmicutes  c__Bacilli  o__Lactobacillales   \n",
       "643   d__Bacteria  p__Firmicutes  c__Bacilli  o__Lactobacillales   \n",
       "1782  d__Bacteria  p__Firmicutes  c__Bacilli  o__Lactobacillales   \n",
       "1322  d__Bacteria  p__Firmicutes  c__Bacilli  o__Lactobacillales   \n",
       "2250  d__Bacteria  p__Firmicutes  c__Bacilli  o__Lactobacillales   \n",
       "\n",
       "                   Family                   Genus                     Species  \\\n",
       "1545  f__Lactobacillaceae  g__Limosilactobacillus                         s__   \n",
       "643   f__Lactobacillaceae  g__Limosilactobacillus                         s__   \n",
       "1782  f__Lactobacillaceae  g__Limosilactobacillus                         s__   \n",
       "1322  f__Lactobacillaceae  g__Limosilactobacillus                         s__   \n",
       "2250  f__Lactobacillaceae            g__Weissella  s__Weissella thailandensis   \n",
       "\n",
       "     sample  cluster  \n",
       "1545    S45      105  \n",
       "643     S25      105  \n",
       "1782    S49      105  \n",
       "1322    S39      105  \n",
       "2250     S8      119  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Lactobacillaceae_df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "144565b8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
